lib/config/Configuration.js

```diff
@@ -0,0 +1,332 @@
+	 * @returns {string}
+	 */
+	getCacheMode() {
+		return this.#cacheMode;
+	}
+
+	/**
+	 * .ui5 home direcotry
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getUi5HomeDir() {
+		return this.#ui5HomeDir;
+	}
+
+	/**
+	 * SNAPSHOTs URL
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getSnapshotEndpointUrl() {
+		return this.#snapshotEndpointUrl;
+	}
+
+	/**
+	 * Directory to store framework artifacts and metadata in.
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getFrameworkDir() {
+		return this.#frameworkDir;
+	}
+
+	/**
+	 * Directory to store artifacts in.
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getArtifactsDir() {
+		return this.#artifactsDir;
+	}
+
+	/**
+	 * Directory to packages in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getPackagesDir() {
+		return this.#packagesDir;
+	}
+
+	/**
+	 * Directory to store metadata in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getMetadataDir() {
+		return this.#metadataDir;
+	}
+
+	/**
+	 * Directory to store staging artifacts in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getStagingDir() {
+		return this.#stagingDir;
+	}
+
+	/**
+	 * Lockfiles directory
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getLockDir() {
+		return this.#lockDir;
+	}
+
+	/**
+	 * Cache directory
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getCacheDir() {
+		return this.#cacheDir;
+	}
+
+	/**
+	 * @public
+	 * @returns {object} The configuration in a JSON format
+	 */
+	toJSON() {
+		return {
+			cwd: this.#cwd,
+			sources: this.#sources,
+			version: this.#version,
+			cacheMode: this.#cacheMode,
+			ui5HomeDir: this.#ui5HomeDir,
+			snapshotEndpointUrl: this.#snapshotEndpointUrl,
+			frameworkDir: this.#frameworkDir,
+			artifactsDir: this.#artifactsDir,
+			packagesDir: this.#packagesDir,
+			metadataDir: this.#metadataDir,
+			stagingDir: this.#stagingDir,
+			lockDir: this.#lockDir,
+			cacheDir: this.#cacheDir,
+		};
+	}
+}
+
+export default Configuration;
+
+/**
+ * Create Configuration from JSON file
+ *
+ * @public
+ * @static
+ * @param {string} [filePath="~/.ui5rc"] Path to configuration JSON file
+ * @returns {@ui5/project/config/Configuration} Configuration instance
+ */
+export async function fromFile(filePath) {
+	filePath = filePath || path.join(os.homedir(), ".ui5rc");
+
+	const {default: fs} = await import("graceful-fs");
+	const {promisify} = await import("node:util");
+	const readFile = promisify(fs.readFile);
+	let config;
+	try {
+		const fileContent = await readFile(filePath);
+		config = JSON.parse(fileContent);
+	} catch (err) {
+		if (err.code === "ENOENT") { // "File or directory does not exist"
+			config = {};
+		} else {
+			throw err;
+		}
+	}
+	return new Configuration(config);
+}
+
+/**
+ * Save Configuration to a JSON file
+ *
+ * @public
+ * @static
+ * @param {string} [filePath="~/.ui5rc"] Path to configuration JSON file
+ * @param {@ui5/project/config/Configuration} config Configuration to save
+ */
+export async function saveConfig(filePath, config) {
```

Comment:
It's clear from the [RFC](https://github.com/SAP/ui5-tooling/blob/rfc-configuration/rfcs/0013-configuration.md#changing-configuration) that this should be part of the CLI.

Would be removed.

Reply:
Given the latest discussions and the future integration with Snapshot Consumption, it's worth it that the Config object would be able to read and write into the `.ui5rc` file. 

The core reasoning behind that is the following scenario:

1. There's an exported static function that reads the file, creates and returns the Configuration object.

2. Writing in `.ui5rc` is a bit more complex: if no `.ui5rc` config is provided, then the Resolver checks for env variable, but there's a last resort if none of these is provided (new project setup, for example). In such a case, it's worth checking for `settings.xml` in case the end-user has maven installed on their machine. Then we need to read the `settings.xml`, extract the URL and save it to `.ui5rc`

We have decided to give it a try. Furthermore, the Configuration object is closed for now and we could eventually revise it